src/client/ws/client.rs

```diff
@@ -36,6 +39,56 @@ use futures::{
 };
 use std::{collections::HashMap, io, marker::PhantomData};
 
+enum InternalChannelSender<T> {
+	AllowLosses(RingSender<T>),
+	BlockWhenFull(mpsc::Sender<T>),
+}
+
+impl<T> InternalChannelSender<T> {
+	fn send(self, data: T) -> Result<(), Error> {
```

Comment:
TODO: rename to `send_non_blocking` and provide an additional `send_async` or something